shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -547,18 +547,29 @@ private void updateViewportMetrics() {
     private void handlePlatformMessage(String channel, String message, final int responseId) {
         OnMessageListener listener = mOnMessageListeners.get(channel);
         if (listener != null) {
-            nativeInvokePlatformMessageResponseCallback(mNativePlatformView, responseId, listener.onMessage(this, message));
+            String response;
+            try {
+                response = listener.onMessage(this, message);
+            } catch (Exception ex) {
+                Log.e(TAG, "Exception in message listener", ex);
+                return;
+            }
+            nativeInvokePlatformMessageResponseCallback(mNativePlatformView, responseId, response);
```

Comment:
I case of an exception, wont the platform side keep holding onto whatever resources are needed to service the response ID?